src/app/icd/ICDMonitoringTable.cpp

```diff
@@ -99,26 +187,29 @@ CHIP_ERROR ICDMonitoringTable::Set(uint16_t index, const ICDMonitoringEntry & en
     VerifyOrReturnError(index < this->Limit(), CHIP_ERROR_INVALID_ARGUMENT);
     VerifyOrReturnError(kUndefinedNodeId != entry.checkInNodeID, CHIP_ERROR_INVALID_ARGUMENT);
     VerifyOrReturnError(kUndefinedNodeId != entry.monitoredSubject, CHIP_ERROR_INVALID_ARGUMENT);
-    VerifyOrReturnError(entry.key.size() == ICDMonitoringEntry::kKeyMaxSize, CHIP_ERROR_INVALID_ARGUMENT);
-    VerifyOrReturnError(nullptr != entry.key.data(), CHIP_ERROR_INVALID_ARGUMENT);
     ICDMonitoringEntry e(this->mFabric, index);
     e.checkInNodeID    = entry.checkInNodeID;
     e.monitoredSubject = entry.monitoredSubject;
-    e.key              = entry.key;
     e.index            = index;
+    memcpy(e.key.AsMutable<Crypto::Aes128KeyByteArray>(), entry.key.As<Crypto::Aes128KeyByteArray>(),
+           sizeof(Crypto::Aes128KeyByteArray));
+
     return e.Save(this->mStorage);
 }
 
 CHIP_ERROR ICDMonitoringTable::Remove(uint16_t index)
 {
-    ICDMonitoringEntry entry(this->mFabric, index);
+    ICDMonitoringEntry entry(mSymmetricKeystore, this->mFabric);
+
+    // Retrieve entry and delete the key first as to not
```

Comment:
I think down below where we do the `entry.Delete(this->mStorage)` we should have a comment that calling `entry.Delete` does not do any key deletion.

Reply:
done